server/src/main/java/org/apache/druid/server/coordinator/rules/Rule.java

```diff
@@ -50,4 +50,12 @@ public interface Rule
   boolean appliesTo(Interval interval, DateTime referenceTimestamp);
 
   void run(DataSegment segment, SegmentActionHandler segmentHandler);
+
+  /**
+   * @return Whether a segment that matches this rule needs to be loaded on a tier.
+   */
+  default boolean shouldSegmentBeLoaded()
+  {
+    return false;
+  }
```

Comment:
This method is not really needed and could be left in LoadRule itself. In `DataSourceResource`, you can ask cast the rule to `rule` to `LoadRule` and then call `shouldSegmentBeLoaded`.

Reply:
can you also add a comment that this method can be used in making handoff decisions?